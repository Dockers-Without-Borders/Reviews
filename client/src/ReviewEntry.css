.reviewEntry_container {
  display: grid;
  grid-gap: 0px;
  grid-template-columns: 200px 200px 200px;
  background-color: #fff;
  grid-row: auto;
    
}
.box {
  border-radius: 0px;
  padding: 10px;
  font-size: 150%;
  margin: 2px;

}

.user {
  grid-column: 1 / 2;
  grid-row: 1;
  background-color: white;
  width: 200px;

}
.review {
  grid-column: 2 /4;
  grid-row: 1 / 3;
  background-color: magenta;
}
.empty {
  grid-column: 1 ;
  grid-row: 2 ;
}

.userPic_container {
  position: relative;
  top: 0px;
  width: 60px;
  height: 60px;
  display: inline-block;
}
.userPic {
  max-width: 100%;
  /* max-height: 100%; */
}

.userInfo_container {
  display: inline-block;
  font-size: 12px;;
}

.hover_container {
  min-height: 200px;
}

.userName {
  font-size: 20;
  font-family: 'Helvetica Neue,Helvetica,Arial,sans-serif';
}