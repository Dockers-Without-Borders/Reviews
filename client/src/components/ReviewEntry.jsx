import React from 'react';
import styles from '../ReviewEntry.css';
import Stars from './Stars.jsx';
import ReviewPictures from './ReviewPictures.jsx'
import UserInfo from './UserInfo.jsx';
import PopoverPics from './PopoverPics.jsx';

class ReviewEntry extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      hover: false,
      user: {},
      review: {},
    }
  }

  componentDidMount() {
    let data = this.props.review;
    let userInfo = {
      name: data.name,
      location: data.location,
      elite: data.elite,
      friend: data.friends,
      numPics: data.numPics,
      picture: data.picture,
      reviews: data.numReviews,
      date: data.date.split('T')[0],
    };

    this.setState({
      user: userInfo
    });
  };

  hoverChange() {
    this.setState({
      hover: !this.state.hover
    });
  };

  
  showPictures() {
    if(this.props.review.links !== null) {
      return <ReviewPictures links={this.props.review.links}/>
    }
  }

  render() {
    return (
      <div className={styles.reviewEntry_container} onMouseEnter={this.hoverChange.bind(this)} onMouseLeave={this.hoverChange.bind(this)}>
        <UserInfo userInfo={this.state.user} hover={this.state.hover}/>
        <div className={styles.review}>
          <div className={styles.starDate}>
            <div>
              <Stars stars={this.props.review.stars}/> 
            </div>
            <div>
              {this.state.user.date}
            </div>
          </div>
          <div>{this.props.review.review}</div>
          {this.showPictures()}
        </div>
      </div>
  )}
}

export default ReviewEntry;