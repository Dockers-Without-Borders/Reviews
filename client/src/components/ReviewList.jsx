import React from 'react';
import ReviewEntry from './ReviewEntry.jsx';


class ReviewList extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      
    }
  }

  render() {
    return (
    <div className='review_container'>
      {this.props.reviews.map(review => <ReviewEntry review={review} hover={this.props.hover}/>)}
    </div>
    )}
}

export default ReviewList;
