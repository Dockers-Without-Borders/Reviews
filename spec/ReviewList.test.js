import React from 'react';
import { shallow } from 'enzyme';

import ReviewList from '../client/src/components/ReviewList';
import { exportAllDeclaration } from '@babel/types';
import ReviewEntry from '../client/src/components/ReviewEntry';

describe('<ReviewList />', () => {
  it('Should render component ReviewList', () => {
    const wrapper = shallow(<ReviewList />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<ReviewList/>', () => {
  it('renders two <ReviewsEntry/>', () => {
    const data = [{'test':'test'}, {'test':'test'}];
    const wrapper = shallow(<ReviewList data={data}/>);
    exportAllDeclaration(wrapper.find(ReviewEntry)).to.have.lengthOf(2);
  });
});