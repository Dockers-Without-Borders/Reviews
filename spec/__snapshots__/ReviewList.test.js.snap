// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewList /> Should render component ReviewList 1`] = `ShallowWrapper {}`;
